app/src/main/java/ml/docilealligator/infinityforreddit/markdown/MarkdownUtils.java

```diff
@@ -0,0 +1,102 @@
+package ml.docilealligator.infinityforreddit.markdown;
+
+import android.content.Context;
+import android.text.util.Linkify;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.commonmark.ext.gfm.tables.TableBlock;
+
+import io.noties.markwon.Markwon;
+import io.noties.markwon.MarkwonPlugin;
+import io.noties.markwon.ext.strikethrough.StrikethroughPlugin;
+import io.noties.markwon.html.HtmlPlugin;
+import io.noties.markwon.html.tag.SuperScriptHandler;
+import io.noties.markwon.inlineparser.AutolinkInlineProcessor;
+import io.noties.markwon.inlineparser.BangInlineProcessor;
+import io.noties.markwon.inlineparser.HtmlInlineProcessor;
+import io.noties.markwon.inlineparser.MarkwonInlineParserPlugin;
+import io.noties.markwon.linkify.LinkifyPlugin;
+import io.noties.markwon.movement.MovementMethodPlugin;
+import io.noties.markwon.recycler.MarkwonAdapter;
+import io.noties.markwon.recycler.table.TableEntry;
+import io.noties.markwon.recycler.table.TableEntryPlugin;
+import me.saket.bettermovementmethod.BetterLinkMovementMethod;
+import ml.docilealligator.infinityforreddit.R;
+import ml.docilealligator.infinityforreddit.customviews.CustomMarkwonAdapter;
+
+public class MarkdownUtils {
+    /**
+     * Creates a Markwon instance with all the plugins required for processing Reddit's markdown.
+     * @return configured Markwon instance
+     */
+    @NonNull
+    public static Markwon createFullRedditMarkwon(@NonNull Context context,
+                                                  @NonNull MarkwonPlugin miscPlugin,
+                                                  int markdownColor,
+                                                  int spoilerBackgroundColor,
+                                                  @Nullable BetterLinkMovementMethod.OnLinkLongClickListener onLinkLongClickListener) {
+        return Markwon.builder(context)
+                .usePlugin(MarkwonInlineParserPlugin.create(plugin -> {
+                    plugin.excludeInlineProcessor(AutolinkInlineProcessor.class);
+                    plugin.excludeInlineProcessor(HtmlInlineProcessor.class);
+                    plugin.excludeInlineProcessor(BangInlineProcessor.class);
+                    plugin.addInlineProcessor(new SuperscriptInlineProcessor());
+                }))
+                .usePlugin(HtmlPlugin.create(plugin -> {
+                    plugin.excludeDefaults(true).addHandler(new SuperScriptHandler());
+                }))
+                .usePlugin(miscPlugin)
+                .usePlugin(SpoilerParserPlugin.create(markdownColor, spoilerBackgroundColor))
```

Comment:
Now that I know how `SpoilerSpan` works, I think that `spoilerBackgroundColor` calculation should be done inside `SpoilerSpan` because it won't work if the background color doesn't match text color